portable/GCC/ARM_CRx_MPU/port.c

```diff
@@ -0,0 +1,799 @@
+
+        if( NULL == xTaskMPUSettings )
+        {
+            xAccessGranted = pdFALSE;
+        }
+        else if( xTaskMPUSettings->ulTaskFlags & portTASK_IS_PRIVILEGED_FLAG )
+        {
+            /* If a task is privileged it is assumed that it can access the buffer */
+            xAccessGranted = pdTRUE;
+        }
+        else
+        {
+            uint32_t ulRegionIndex = 0x0UL;
+            do
+            {
+                xTaskMPURegion = &( xTaskMPUSettings->xRegion[ ulRegionIndex++ ] );
+                xAccessGranted = prvTaskCanAccessRegion(
+                    xTaskMPURegion,
+                    ( uint32_t ) pvBuffer,
+                    ulBufferLength,
+                    ulAccessRequested
+                );
+            } while( ( pdFALSE == xAccessGranted ) &&
+                     ( ulRegionIndex < portTOTAL_NUM_REGIONS_IN_TCB ) );
+        }
+    }
+    return xAccessGranted;
+}
+
+/*---------------------------------------------------------------------------*/
+
+/** @brief Determine if the FreeRTOS Task was created as a privileged task
+ *
+ * @ingroup MPU Control
+ * @ingroup Task Context
+ *
+ * @return pdTRUE if the Task was created as a privileged task.
+ * pdFALSE if the task was not created as a privileged task.
+ *
+ */
+/* PRIVILEGED_FUNCTION */ BaseType_t xPortIsTaskPrivileged( void )
+{
+    BaseType_t xTaskIsPrivileged = pdFALSE;
+
+    /* Calling task's MPU settings. */
+    const xMPU_SETTINGS * xTaskMpuSettings = xTaskGetMPUSettings( NULL );
+
+    if( ( xTaskMpuSettings->ulTaskFlags & portTASK_IS_PRIVILEGED_FLAG ) ==
+        portTASK_IS_PRIVILEGED_FLAG )
+    {
+        xTaskIsPrivileged = pdTRUE;
+    }
+
+    return xTaskIsPrivileged;
+}
+
+/** @brief Start the FreeRTOS-Kernel's control of Tasks by starting the System Tick
+ * Interrupt.
+ *
+ * @ingroup Scheduler
+ * @return BaseType_t This function is not meant to be returned from.
+ * If it does return it returns pdFALSE to mark that the scheduler could not be started.
+ */
+BaseType_t xPortStartScheduler( void )
+{
+    /* Start the timer that generates the tick ISR. */
+    configSETUP_TICK_INTERRUPT();
+
+    /* Reset the critical section nesting count read to execute the first task. */
+    ulCriticalNesting = 0UL;
+
+    /* Configure the regions in the MPU that are common to all tasks. */
+    prvSetupDefaultMPU();
+
+    xSchedulerRunning = pdTRUE;
+
+    /* Load the context of the first task, starting the FreeRTOS-Scheduler's control. */
+    vPortStartFirstTask();
+
+    /* Will only get here if vTaskStartScheduler() was called with the CPU in
+     * a non-privileged mode or the binary point register was not set to its lowest
+     * possible value. prvTaskExitError() is referenced to prevent a compiler
+     * warning about it being defined but not referenced in the case that the user
+     * defines their own exit address. */
+    ( void ) prvTaskExitError();
+    return 0;
```

Comment:
Minor nitpick, should return pdFALSE to match the comment above

Reply:
Return value adjusted